### PR TITLE
Tab names for slice tabs as file names.

### PR DESCRIPTION
**Pull request by maharshi-gor**
### Changes
This PR introduces a data field saying file name

### The behavior to observe: 
- On the page a field `Filename` got added with the file name next to it.
-
Screenshot from 2024-01-30 15-40-11

- I have updated all the test cases accordingly to take care of file while unpacking the images data.

### CODE REVIEW

File: dipy/viz/horizon/app.py [hunk 1]
```diff
@@ -443,7 +442,8 @@ def build_show(self, scene):
             img_count = 0
             synchronize_slices = check_img_shapes(self.images)
             for img in self.images:
-                data, affine = img
+                data, affine, fname = img
+                fname = fname.split("/")[-1]
```
Comment (maharshi-gor) [inline dipy/viz/horizon/app.py:446]: splitting the file path and extracting the name

Comment (skoudoro) [reply dipy/viz/horizon/app.py:446]: Are you sure it works on windows? I will be surprised because they usually use `\\`.  I recommend you to use `os.path.basename`. You can also look at the `Pathlib` python module.

File: dipy/viz/horizon/util.py [hunk 2]
```diff
@@ -1,14 +1,69 @@
...
+    extension : string
+        Extension that needs to be removed from file name. If the '.' is not
+        included in the extension it will automatically added.
+
+    Returns
+    -------
+    string
+        File name with the extension in it.
+    """
+    if extension[0] != '.':
```
Comment (skoudoro) [inline dipy/viz/horizon/util.py:65]: `if not extension.startswith('.')`  instead

File: dipy/viz/horizon/tab/base.py [hunk 3]
```diff
@@ -103,16 +106,29 @@ def __init__(self, tabs, win_size, synchronize_slices=False):
...
+        self._scene.add(self._tab_ui)
+
+        for tab_id, tab in enumerate(tabs):
+            self._handle_title_overflow(
+                tab.name,
+                self._tab_ui.tabs[tab_id].text_block
+            )
+
+    def _handle_title_overflow(self, title_text, title_block):
+        tab_text = remove_extension(title_text, 'nii')
```
Comment (skoudoro) [inline dipy/viz/horizon/tab/base.py:125]: Not sure you need a function for that....    

`tab_text = title_text.split('.', 1)[0]` should just be enough.....

in `pathlib` module, you also have `suffixes()`

File: dipy/viz/horizon/tab/base.py [hunk 4]
```diff
@@ -103,16 +106,29 @@ def __init__(self, tabs, win_size, synchronize_slices=False):
...
 
+        self._scene.add(self._tab_ui)
+
+        for tab_id, tab in enumerate(tabs):
+            self._handle_title_overflow(
+                tab.name,
+                self._tab_ui.tabs[tab_id].text_block
+            )
+
+    def _handle_title_overflow(self, title_text, title_block):
```
Comment (skoudoro) [inline dipy/viz/horizon/tab/base.py:124]: Can you add a docstring.

**Review by skoudoro — COMMENTED** [review]
Thanks @maharshi-gor,

See below some comments to fix before I try on my latptop.

Can you also confirm that this work only at the startup. if we resize the window, it does not refresh the filename, right ?

File: dipy/viz/horizon/app.py [hunk 5]
```diff
@@ -485,19 +483,13 @@ def build_show(self, scene):
...
-
         if len(self.__tabs) > 0:
-            self.__tab_mgr = TabManager(self.__tabs, self.__win_size,
-                                        synchronize_slices)
-
-            def tab_changed(actors):
-                for act in actors:
-                    scene.rm(act)
-                    scene.add(act)
+            def get_size(actor):
```
Comment (skoudoro) [inline dipy/viz/horizon/app.py:487]: Why did you add this function? where do you use it ?

**Comment by maharshi-gor** [comment]
> Thanks @maharshi-gor,
> 
> See below some comments to fix before I try on my latptop.
> 
> Can you also confirm that this work only at the startup. if we resize the window, it does not refresh the filename, right ?

The resize window will required more changes. The current PR only targets the startup.

**Review by skoudoro — CHANGES_REQUESTED** [review]
Hi @maharshi-gor,

Can you merge master on this PR or rebase to fix the conflicts.

is it normal to have 3 dots (ellipsis) at the beginning of the filename ? it is weird, see below the weird filename:

image



EDIT: the name should be `mni_icbm152_t1_tal_nlin_asym_09c.nii`

**Review by skoudoro — APPROVED** [review]
ok, all good, thanks @maharshi-gor!

merging
